website/project/views/mailing_list.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+from website.project.decorators import (
+    must_be_valid_project,
+    must_have_permission,
+    must_not_be_registration,
+)
+from website.util.permissions import ADMIN
+
+from website.project.mailing_list import route_message
+
+
+###############################################################################
+# View Functions
+###############################################################################
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def get_node_mailing_list(node, auth, **kwargs):
+    return format_node_data_recursive([node], auth.user)
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def enable_mailing_list(node, **kwargs):
+    node.mailing_enabled = True
+    node.save()
+
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def disable_mailing_list(node, **kwargs):
+    node.mailing_enabled = False
+    node.save()
+
+route_message = route_message
+
+def format_node_data_recursive(nodes, user):
+    items = []
+
+    for node in nodes:
+
+        can_read = node.has_permission(user, 'read')
+        can_read_children = node.has_permission_on_children(user, 'read')
+
+        if not can_read and not can_read_children:
+            continue
+
+        children = []
+        # List project/node if user has at least 'read' permissions (contributor or admin viewer) or if
+        # user is contributor on a component of the project/node
+
+        if can_read_children:
+            children.extend(format_node_data_recursive(
+                [
+                    n
+                    for n in node.nodes
+                    if n.primary and
+                    not (n.is_deleted or n.is_registration)
+                ],
+                user
+            ))
+
+        item = {
+            'node': {
+                'id': node._id,
+                'url': node.url if can_read else '',
+                'title': node.title if can_read else 'Private Project',
+                'mailing_list': ('enabled' if node.mailing_enabled else 'disabled') if can_read else None
+            },
+            'children': children,
+            'kind': 'folder' if node.is_folder else 'node',
```

Comment:
Let's drop back to the expanded `node__parent` condition, since, as you point out at https://github.com/CenterForOpenScience/osf.io/pull/5033#discussion_r54162986, that's what is used in other treebeards. We should change those all at once. Since we aren't confident of the change (https://github.com/CenterForOpenScience/osf.io/pull/5033#commitcomment-16541010), let's not to add risk to this mega mondo PR. Let's do that in a separate (small) PR instead. Please either make that PR now or reticket in Jira.